src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -64,6 +65,14 @@ def find_vdev(pool, vname):
         children += list(child.children)
 
 
+def libzfs_errcode_mapping(e):
+    return {
+        ZFSError.NOSPC: errno.ENOSPC,
+        ZFSError.NOREPLICAS: errno.ENOTRECOVERABLE,
+        ZFSError.NODEVICE: errno.ENOENT,
+    }.get(e.code) or e.code
```

Comment:
I don't like that we are mixing `errno` and `zfs_error` (which AFAIK do not match `errno`s), can it be `errno.EFAULT` by default instead?

Reply:
Its either that or have all errors specified in middleware (or they shown as EUNKNOWN) which I think its worse.

Defaulting to e.code at least let us identify what the original error code was and fix it up.

Thoughts?

Reply:
Can we make it `300 + e.code` for example? And unparse it back in `ErrnoMixin`

Reply:
I dont understand how that helps? e.code for ZFS are already in the 2000+ range.

Reply:
I want to isolate them from errno codes so they don't intersect. If they are already that way, fine.
But `ErrnoMixin` part seems useful to me nevertheless.

Reply:
Still not sure what you mean, are you talking about doing the inverse of `ZFSError.NODEVICE: errno.ENOENT` ?

And if so, why?

Reply:
I want to do this in `ErrnoMixin`:
```

    @classmethod
    def _get_errname(cls, code):
        if 2000 <= code <= 2100:
            return 'EZFS' + ZFSError(code).name    
```

Reply:
Ok, thats much smarter ;)